core/src/main/java/org/infinispan/remoting/transport/impl/SingletonMapResponseCollector.java

```diff
@@ -13,18 +13,33 @@
 import org.infinispan.remoting.rpc.ResponseMode;
 import org.infinispan.remoting.transport.Address;
 import org.infinispan.remoting.transport.ValidSingleResponseCollector;
+import org.infinispan.remoting.transport.jgroups.JGroupsTransport;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
 
 /**
  * Response collector supporting {@link JGroupsTransport#invokeRemotelyAsync(Collection, ReplicableCommand, ResponseMode, long, ResponseFilter, DeliverOrder, boolean)}.
  *
  * @author Dan Berindei
- * @since 9.1
+ * @since 9.2
  */
-class SingletonMapResponseCollector
+public class SingletonMapResponseCollector
       extends ValidSingleResponseCollector<Map<Address, Response>> {
+   private static final Log log = LogFactory.getLog(SingletonMapResponseCollector.class);
+   private static final SingletonMapResponseCollector VALID_ONLY = new SingletonMapResponseCollector(false);
+   private static final SingletonMapResponseCollector IGNORE_LEAVERS = new SingletonMapResponseCollector(true);
+
    private final boolean ignoreLeavers;
```

Comment:
I would have done the same thing as in `MapResponseCollector`

Reply:
It makes sense to remove the field in `MapResponseCollector` because it saves some memory, but here we don't allocate new instances for each request.